core/templates/services/compute-graph.service.ts

```diff
@@ -79,13 +79,13 @@ export class ComputeGraphService {
   _computeBfsTraversalOfStates(
       initStateId: string, states: States, sourceStateName: string): string[] {
     let stateGraph = this._computeGraphData(initStateId, states);
-    let stateNamesInBfsOrder = [];
-    let queue = [];
-    let seen = {};
+    let stateNamesInBfsOrder: string[] = [];
+    let queue: string[] = [];
+    let seen: Record<string, boolean> = {};
     seen[sourceStateName] = true;
     queue.push(sourceStateName);
     while (queue.length > 0) {
-      let currStateName = queue.shift();
+      let currStateName = <string>queue.shift();
```

Comment:
Already checking for queue.length to be greater than 0. `.shift()` here won't return undefined